### PR TITLE
Allow add_categories() to accept Series/np.array and make raising on dup...

### LINKED ISSUE #9927: API/BUG: awkward syntax to add categories to a Categorical
from [SO](http://stackoverflow.com/questions/29709918/pandas-and-category-replacement/29712287#29712287)

- the ``.add_categories`` should be able to take an ``Index``/ndarray, ATM it must be converted to a list
- should be a keyword for something like: add any additional categories that I am passing to you (even if the current ones are duplicates, just add these to the end). e.g.

```
In [147]: s = pd.Categorical(list('aabbcd'))

In [148]: s2 = list('aabbcdefg')

In [149]: s
Out[149]: 
[a, a, b, b, c, d]
Categories (4, object): [a, b, c, d]

In [150]: s.add_categories(s.categories.sym_diff(Index(s2)).tolist())
Out[150]: 
[a, a, b, b, c, d]
Categories (7, object): [a, b, c, d, e, f, g]
```

I would ideally just like to say:

```
s.add_categories(s2, take_new=True)
```

(maybe not the best keyword, but something like this)

### PR DESCRIPTION
**Pull request by mortada**
...licates optional, fixes #9927

### CODE REVIEW

**Comment by jreback** [comment]
@mortada ok, for now pls take out the ``raise_on_duplicates`` kw and leave the rest.

File: pandas/tests/test_categorical.py [hunk 1]
```diff
@@ -727,6 +727,19 @@ def f():
...
         self.assertRaises(ValueError, f)
 
+        # GH 9927
+        # with raise_on_duplicates=False
+        cat = Categorical(list("abc"), ordered=True)
+        res = cat.add_categories(list("cd"), raise_on_duplicates=False)
+        expected = Categorical(list("abc"), categories=list("abcd"), ordered=True)
+        self.assert_categorical_equal(res, expected)
+
+        # with Series or np.array
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: test reg Index & list as well

**Comment by mortada** [comment]
@jreback  it's updated

**Comment by jreback** [comment]
@mortada thanks!

**Pull request merged**